books_project/books_app/components/book_component.py

```diff
@@ -0,0 +1,63 @@
+from books_app.repositories.book_repository import BookRepository
+from rest_framework.exceptions import NotFound, ValidationError
+
+from books_app.repositories.author_repository import AuthorRepository
+
+class BookComponent:
+
+    def create_book(self, title, published_year, genr, isbn, author_id):
+        exists = BookRepository.check_isbn_exist(isbn)
+        author = AuthorRepository.get_author(author_id=author_id)
+        if not author:
+            raise NotFound("author not found", code=404)
+        elif title != "":
+            raise ValidationError(detail="title is required", code=400)
+        elif exists:
+            raise ValidationError(detail="isbn is already exists", code=400)
+        BookRepository.create_book(title, published_year, genr, isbn, author=author)
+
+        
+    def update_book(self, book_id, title, published_year, genr, isbn, author_id, **kwargs):
+        book_data = {
+        'id': id,
+        'title': title,
+        'published_year': published_year,
+        'genre': genr,
+        'isbn': isbn,
+        }
```

Comment:
indentation. add extra spacing inside the dict to follow python contentions
```
book_data = {
    'id': book_id,
    'title': title,
    'published_year': published_year,
    'genre': genr,
    'isbn': isbn,
}
```
also just like before, we don't have "id" here, you need book_id.